### PR TITLE
feat!: init doctor command

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
Initializes the doctor singleton in the oclif plugin init hook
### What issues does this PR fix or reference?
@W-12647786@

### CODE REVIEW

File: test/commands/doctor.test.ts [hunk 1]
```diff
@@ -51,35 +66,36 @@ describe('Doctor Command', () => {
   let promptStub: sinon.SinonStub;
   let openStub: sinon.SinonStub;
 
-  oclifConfigStub = fromStub(
-    stubInterface<Config>(sandbox, {
-      pjson: {
-        engines: {
-          node: 'node-v16.17.0',
-        },
+  oclifConfig = {
```
Comment (shetzel) [inline test/commands/doctor.test.ts:69]: If you use the `fromStub(stubInterface<>)` pattern you won't need to do the `as unknown as Config;`. That's the super cool thing about using those utilities.

File: test/diagnostics.test.ts [hunk 2]
```diff
@@ -44,22 +59,17 @@ describe('Diagnostics', () => {
...
     lifecycleEmitSpy = spyMethod(sandbox, lifecycle, 'emit');
-    oclifConfigStub = fromStub(
-      stubInterface<Config>(sandbox, {
-        name: 'sfdx-cli',
-        version: '7.160.0',
-        pjson: {
-          engines: {
-            node: 'node-v16.17.0',
-          },
+    oclifConfig = {
```
Comment (shetzel) [inline test/diagnostics.test.ts:62]: Same here.  `fromStub(stubInterface)` handles proper typing of stubs.

**Review by shetzel — CHANGES_REQUESTED** [review]
I would put the `fromStub(stubInterface)` calls back.

File: test/commands/doctor.test.ts (hunk 1 above)
Comment (peternhale) [reply test/commands/doctor.test.ts:69]: @shetzel That's how I started, but in the tests when I would override oclifConfig.versionDetails = {...} the change never appeared.

**Comment by mshanemc** [comment]
this has gotten pretty stale.  probably better to start a new one if needed

**Pull request closed without merging**